File: item-list/src/test/java/com/gurunars/item_list/TestItem.java
package com.gurunars.item_list;

class TestItem extends Item<Integer> {

    TestItem(int id, Integer payload) {
        super(null, id, payload);
    }

    @Override
    public String toString() {
        return "new TestItem(" + getId() + ", " + getPayload() + ")";
    }

}

File: item-list/src/test/kotlin/com/gurunars/item_list/TestItem.kt
package com.gurunars.item_list

internal class TestItem(override val id: Long, private val version: Int) : Item {

    enum class Type {ONE}

    override val type: Enum<*>
        get() = Type.ONE

    override fun toString(): String {
        return "new TestItem($id, $version)"
    }

    override fun equals(other: Any?): Boolean {
        if (other == null || other !is TestItem) {
            return false
        }
        return id == other.id && version == other.version
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + version
        return result
    }
}
